src/video.c

```diff
@@ -876,7 +942,20 @@ static void stream_recv_handler(const struct rtp_header *hdr,
 	if (lostc)
 		request_picture_update(&v->vrx);
 
-	(void)video_stream_decode(&v->vrx, hdr, mb);
+	q = mem_zalloc(sizeof(struct vidqdec), vidqdec_destructor);
+	if (!q)
+		return;
+
+	q->hdr = mem_ref((void *)hdr);
+	q->mb  = mem_ref(mb);
+
+	update_rtp_timestamp(&v->vrx.ts_recv, hdr->ts);
+
+	mtx_lock(v->vrx.lock_rx);
+	q->vrxc = v->vrx; /* copy vrx */
+	list_append(&v->vrx.recvq, &q->le, q);
```

Comment:
the queue should have an upper limit for maximum number of entries.

if the decode thread is slower than the main thread, it will start to accumulate
entries in the queue. After some time this will lead to extra delay for the video stream.